kube-controllers/pkg/controllers/node/hostendpoints.go

```diff
@@ -389,21 +532,25 @@ func (c *autoHostEndpointController) hostendpointNeedsUpdate(current *api.HostEn
 	return false
 }
 
-// updateHostendpoint updates the current hostendpoint so that it matches the
-// expected hostendpoint.
-func (c *autoHostEndpointController) updateHostendpoint(current *api.HostEndpoint, expected *api.HostEndpoint) error {
-	if c.hostendpointNeedsUpdate(current, expected) {
+// updateHostEndpoint updates the current HostEndpoint so that it matches the expected HostEndpoint.
+func (c *autoHostEndpointController) updateHostEndpoint(current *api.HostEndpoint, expected *api.HostEndpoint) error {
+	if c.hostEndpointNeedsUpdate(current, expected) {
 		logrus.WithField("hep.Name", current.Name).Debug("hostendpoint needs update")
-		expected.ResourceVersion = current.ResourceVersion
-		expected.ObjectMeta.CreationTimestamp = current.ObjectMeta.CreationTimestamp
-		expected.ObjectMeta.UID = current.ObjectMeta.UID
-
-		rlKey := rateLimiterItemKey{Type: RateLimitCalicoUpdate, Name: current.Name}
-		time.Sleep(c.rl.When(rlKey))
-		_, err := c.client.HostEndpoints().Update(context.Background(), expected, options.SetOptions{})
-		if err == nil {
-			c.rl.Forget(rlKey)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		latestHostEndpoint, err := c.client.HostEndpoints().Get(ctx, expected.Name, options.GetOptions{})
+		if err != nil {
+			return err
 		}
+
+		expected.ResourceVersion = latestHostEndpoint.ResourceVersion
+		expected.ObjectMeta.CreationTimestamp = latestHostEndpoint.ObjectMeta.CreationTimestamp
+		expected.ObjectMeta.UID = latestHostEndpoint.ObjectMeta.UID
+
+		ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		_, err = c.client.HostEndpoints().Update(ctx, expected, options.SetOptions{})
```

Comment:
One thing we could do it proactively update our cache here - that would do two things:
- Avoid tricky / awkward bugs where we have to wait for an Update from the API server for our in-memory representation to be correct. e.g., in case we want to do any further processing on the HEP after updating it.
- Allow us to filter out updates that we already know about when triggering syncs based on HEP events. i.e., if we get an event but the new HEP is identical to the one in our cache, we don't need to trigger a sync.